fuel/datasets/cifar10.py

```diff
@@ -40,46 +33,29 @@ class CIFAR10(IndexableDataset):
 
     """
     provides_sources = ('features', 'targets')
-    folder = 'cifar10'
-    files = {
-        'train': [os.path.join('cifar-10-batches-py',
-                               'data_batch_{}'.format(i))
-                  for i in range(1, 6)],
-        'test': ['cifar-10-batches-py/test_batch']
-    }
+    filename = 'cifar10.hdf5'
 
     def __init__(self, which_set, flatten=True, **kwargs):
         if which_set not in ('train', 'test'):
             raise ValueError("CIFAR10 only has a train and test set")
 
+        kwargs.setdefault('load_in_memory', True)
         self.which_set = which_set
-        self.flatten = flatten
+        self.reshape = flatten
 
-        super(CIFAR10, self).__init__(OrderedDict(zip(self.provides_sources,
-                                                      self._load_cifar10())),
+        super(CIFAR10, self).__init__(self.data_path, which_set,
                                       **kwargs)
 
-    def load(self):
-        self.indexables = [data[self.start:self.stop] for source, data
-                           in zip(self.provides_sources, self._load_cifar10())
-                           if source in self.sources]
-
-    def _load_cifar10(self):
-        base_path = os.path.join(config.data_path, self.folder)
-        num_examples = 50000 if self.which_set == 'train' else 10000
-        image_shape = (3072,) if self.flatten else (3, 32, 32)
-        images = numpy.zeros((num_examples,) + image_shape,
-                             dtype=config.floatX)
-        labels = numpy.zeros((num_examples, 1), dtype='uint8')
-        for i, fname in enumerate(self.files[self.which_set]):
-            with open(os.path.join(base_path, fname), 'rb') as f:
-                if six.PY3:
-                    batch = cPickle.load(f, encoding='latin1')
-                else:
-                    batch = cPickle.load(f)
```

Comment:
In order to fix the current test failure under Python 3, you should have a look at the above code.